app/code/core/Mage/Adminhtml/controllers/Sales/OrderController.php

```diff
@@ -277,13 +277,23 @@ public function addCommentAction()
                 $notify = isset($data['is_customer_notified']) ? $data['is_customer_notified'] : false;
                 $visible = isset($data['is_visible_on_front']) ? $data['is_visible_on_front'] : false;
 
-                $order->addStatusHistoryComment($data['comment'], $data['status'])
+                $comment = $data['comment'];
+
+                $session = Mage::getSingleton('admin/session');
+                $username = $session->getUser()->getUsername();
+                
+                if (!$notify && !$visible) {
```

Comment:
Lines 282 and 283 should be after the `if` statement as they are only needed in `if`:

```
if (!$notify && !$visible) {
    $session = Mage::getSingleton('admin/session');
    $username = $session->getUser()->getUsername();
    $comment .= (!issempty($username)) ? " ({$username})" : ' (system)';
}